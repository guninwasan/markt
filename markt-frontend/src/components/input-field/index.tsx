export { LoginInputField, RegisterInputField } from "./input-field";
