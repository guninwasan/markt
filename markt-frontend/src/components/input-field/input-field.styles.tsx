const inputStyles = {
    width: '29.625rem',
    height: '2.625rem',
    flexShrink: 0,
    borderRadius: '0.3125rem',
    background: '#D9D9D9',
    boxShadow: '0px 4px 4px 0px rgba(0, 0, 0, 0.25)',
    padding: '0.5rem',
    border: 'none',
    fontSize: '1rem',
  };

export { inputStyles };