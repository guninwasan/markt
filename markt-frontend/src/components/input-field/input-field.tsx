import React from 'react';

import { inputStyles } from './input-field.styles';

interface LoginInputFieldProps {
    type: string;
    placeholder: string;
    value: string;
    onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
    onBlur: (e: React.FocusEvent<HTMLInputElement>) => void;
    style?: React.CSSProperties; // Add this line
}

interface RegisterInputFieldProps {
    type: string;
    placeholder: string;
    value: string;
    onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
    onBlur: (e: React.FocusEvent<HTMLInputElement>) => void;
    style?: React.CSSProperties;
}

const LoginInputField: React.FC<LoginInputFieldProps> = ({ type, placeholder, onChange, onBlur }) => {
    return <input type={type} placeholder={placeholder} style={inputStyles} onChange={onChange} onBlur={onBlur} />;
};

const RegisterInputField: React.FC<RegisterInputFieldProps> = ({ type, placeholder, onChange, onBlur }) => {
    return <input type={type} placeholder={placeholder} style={inputStyles} onChange={onChange} onBlur={onBlur} />;
};

export { LoginInputField, RegisterInputField };
