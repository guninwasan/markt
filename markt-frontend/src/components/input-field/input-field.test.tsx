import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { InputField } from "../input-field";

describe("InputField", () => {
  it("should render the InputField component", () => {
    render(<InputField />);
    expect(screen.getByPlaceholderText("Enter text")).toBeInTheDocument();
  });

  it("should call onChange when input value changes", () => {
    const handleChange = jest.fn();
    render(<InputField onChange={handleChange} />);
    const input = screen.getByPlaceholderText("Enter text");

    fireEvent.change(input, { target: { value: "test" } });
    expect(handleChange).toHaveBeenCalledWith("test");
  });

  it("should display error message when provided", () => {
    const errorMessage = "This field is required";
    render(<InputField errorMessage={errorMessage} />);
    expect(screen.getByText(errorMessage)).toBeInTheDocument();
  });

  it("should accept and display input value", () => {
    render(<InputField />);
    const input = screen.getByPlaceholderText("Enter text");

    fireEvent.change(input, { target: { value: "test" } });
    expect(input).toHaveValue("test");
  });

  it("should match snapshot", () => {
    const { container } = render(<InputField />);
    expect(container).toMatchSnapshot();
  });
});
