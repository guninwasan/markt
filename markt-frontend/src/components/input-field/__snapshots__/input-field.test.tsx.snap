// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField should match snapshot 1`] = `
<div>
  <div
    class="sc-blHHSb fpPZQa"
  >
    <div
      class="sc-gtLWhw dewetd"
    >
      <input
        class="sc-egkSDF hTIQeX"
        placeholder="Enter text"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
